### DISCUSSION TITLE
Looking for a Blockly-based micropython implementation (IDE)

### DISCUSSION THREAD

**Discussion by glenp68** [micropython/micropython#11637] [General]
HI All

I'm looking for a blockly-based IDE that is based on micropython and allows for OTA to get my daughter into programming an ESP32.

Bipes is great, and fits the brief, but the project seems to have got to 2021 and has stalled with the micropython base becoming outdated.

I would prefer something web-based that kiddo can use on an iPad but an open to others if that's my only choice.

Any suggestions?  

Glen.

**Comment by dhalbert** [comment]
You have probably already looked at these:
https://github.com/microBlock-IDE/microBlock-IDE seems to handle some specific ESP32 boards.
https://mpyblockly.github.io/mpyblockly/ seems brand new.

If you are willing to stray further than plain ESP32 and MicroPython:
https://app.edublocks.org/ does CircuitPython and a number of other platforms. The author just got hired by Anaconda.
https://makecode.org is pretty polished. You can see generated Python or Javascript code, but the main point is block programming. 
https://learn.adafruit.com/quickstart-adafruit-io-wippersnapper/welcome-to-wippersnapper. Wippersnapper (deliberate misspelling) is a no-code IoT solution that runs on a lot of boards. Not block programming per se.

I guest-taught one session of a Girls Who Code course a few years ago. They were learning Javascript on laptops, but it was slow going. I brought in some Circuit Playground Express boards and taught them MakeCode for about 5-10 minutes. They took off from there and had a great time for the next 1.5 hours, using the board features: RGB LEDs, sound input and output, the on-board accelerometer, and light sensor.  I also showed them CircuitPython with Mu, but they liked MakeCode a lot better.

**Comment by bixb922** [comment]
https://oxocard.ch/en/blocklycard/ is for learning

**Comment by mispacek** [comment]
<img width="1426" height="981" alt="image" src="https://github.com/user-attachments/assets/79f9f491-2713-48f1-acd1-f5a64fefe399" />

---

I’m Milan Špaček. I made ESP IDE a free, open source, web-based IDE for ESP8266 / ESP32 / ESP32-C3 / ESP32-S3 / RP2040 (more boards coming). It is based on MicroPython 1.27.
It is great for beginners who want to quickly try sensors, motors, servos, and displays.

Great for RC projects:
✅ Built-in Bluetooth remote controller
✅ ESP-NOW for low-latency wireless control

Sensor support:
✅ DHT11 / DHT22 / AM2302 temperature & humidity
✅ HC-SR04 ultrasonic distance sensor
✅ Analog joysticks
✅ Rotary encoders
✅ TCS34725 RGB color sensor
✅ MPU6050 IMU (gyro + accel)
✅ VL53L0X laser distance sensor

Display support:
✅ SSD1306 / SSD1309 OLED
✅ ST7567 LCD
✅ HD44780 character LCD over I2C (add-on)

LEDs and outputs:
✅ NeoPixel LED strip
✅ Servos (180° and 360°)
✅ DC motors with PWM speed control

Add-ons:
ESP IDE has a simple add-on system. Anyone can add new hardware libraries or new features.

More features are included, and more are coming.

Documentation is still the weak point, but I’m working on it now.

Try it: https://espide.eu/en/index.php

I’d love feedback, help, or ideas to make ESP IDE better for everyone.
